### DISCUSSION TITLE
win下读取pag，保存为png都是黑色

### DISCUSSION THREAD

**Discussion by jackTJT** [Tencent/libpag#1880] [General] [answered]
在使用libpag读取pag时，将每一帧保存为BGRA_8888格式后用opencv保存为png时发现都是黑色
结果图：
<img width="434" alt="b56d35b4498d6e66b9a605119904133c" src="https://github.com/Tencent/libpag/assets/96832322/e59e25ff-a1b7-4646-ae44-c57eb0baa1c1">
读取的pag：
<img width="412" alt="d99bf686d21016f6ebe0184bb77af21a" src="https://github.com/Tencent/libpag/assets/96832322/71db7f7b-21d7-41fd-8dc7-feced43f3416">
相关代码
<img width="485" alt="728f49a0d70a0fd77f760f73652fa276" src="https://github.com/Tencent/libpag/assets/96832322/820d6d13-05bc-4eec-9df1-2ffa692fe5d6">

**Comment by jackTJT** [comment]
打印date时都是0，想问一下该如何获取到正确的数据

**Comment by kevingpqi123** [comment, marked as answer]
这里是数据读取编码 png 的问题，可以到网上搜索下答案：https://blog.csdn.net/weixin_45143156/article/details/109054231

**Reply by jackTJT** [reply]
我这边实验发现调用readPixels给date赋值，打印date时里面数据都是0，所以导致的保存为png是全黑的
